### ISSUE TITLE
0.24: FileNotFoundException on null file

### ISSUE THREAD

**Issue by tedder** [CodeNarc/CodeNarc#113]
I upgraded to 0.24 from 0.23, no other changes, and codenarc fails in the following way:

```
|Compiling 356 source files
..Adding report HtmlReport
Adding report TextReport
Running CodeNarc (file:grails-app/conf/CodeNarcRules.groovy) ...
.FAILED -- java.io.FileNotFoundException: File [null] does not exist or is not accessible
```

I'm happy to do more troubleshooting but I don't know how to turn on verbose output when running Grails. I tried `--verboseCompile` but didn't see anything.

**Comment by tedder** [comment]
If it helps, [here's a version code diff](https://github.com/CodeNarc/CodeNarc/compare/CodeNarc-0.23...CodeNarc-0.24). I looked for "new File" and "File(" and nothing obvious stuck out. My gut says it's in the ReportWriter refactor, but since I don't know how to get a stacktrace from a grails plugin, I don't know how to help further.

**Comment by chrismair** [comment]
I'll take a look. Thanks.

**Comment by chrismair** [comment]
What version of Grails are you using?

I was able to reproduce this issue on Grails 2.2.3, but not with Grails 2.3 (at least not yet).

The current documented requirement is: *Version 0.23 and later of this plugin requires Grails 2.3.0 or later. (Earlier 2.x versions may work, but no guarantees).*  But I know you said that version 0.23 worked fine for you.

**Comment by tedder** [comment]
I'm on 2.5.0.

**Comment by chrismair** [comment]
So far, this seems specific to the *Grails CodeNarc plugin* (not base CodeNarc), so I have opened a corresponding plugin issue: https://github.com/chrismair/GrailsCodeNarcPlugin/issues/6

In the meantime, I was able to work around the issue by adding this to "BuildConfig.groovy":

    // TODO Workaround for https://github.com/chrismair/GrailsCodeNarcPlugin/issues/6
    codenarc.excludeBaseline = ''

**Comment by tedder** [comment]
Thanks for doing the work to narrow it down to the plugin. I added that config to BuildConfig and can confirm it works now.

**Comment by tedder** [comment]
unrelated but not worth an issue yet: the sorting is inefficient for large numbers of errors- I just tried it with 5000. I may try to convert the errors to an array, then use JS to display the table.

**Comment by chrismair** [comment]
Fixed in CodeNarc Grails Plugin version 0.24.1.

See https://github.com/chrismair/GrailsCodeNarcPlugin/issues/6
